lib/internal/main/mksnapshot.js

```diff
@@ -119,10 +120,15 @@ function requireForUserSnapshot(id) {
     err.code = 'MODULE_NOT_FOUND';
     throw err;
   }
-  if (!supportedInUserSnapshot(normalizedId)) {
+  if (isBuildingSnapshot() && !supportedInUserSnapshot(normalizedId)) {
     if (!warnedModules.has(normalizedId)) {
-      process.emitWarning(
-        `built-in module ${id} is not yet supported in user snapshots`);
+      emitWarningSync(
```

Comment:
```suggestion
      // Emit the warning synchronously in case we don't get to process
      // the tick and print it before the unsupported built-in causes a
      // crash.
      emitWarningSync(
```